Sources/NIOHTTP1/HTTPServerPipelineHandler.swift

```diff
@@ -477,6 +527,31 @@ public final class HTTPServerPipelineHandler: ChannelDuplexHandler, RemovableCha
         }
     }
 
+    /// - Returns: True if an error was sent, ie the caller should not continue
+    private func handleConnectionStateAction(
+            context: ChannelHandlerContext,
+            action: ConnectionStateAction,
+            promise: EventLoopPromise<Void>?
+    ) -> Bool {
+        switch action {
+        case .warnPreconditionViolated(let message):
+            self.preconditionFailed(message)
+            let error = ConnectionStateError.preconditionViolated(message: message)
+            self.deliverOneError(context: context, error: error)
+            promise?.fail(error)
+            return true
+        case .forceCloseConnection:
+            let message = "The connection has been forcefully closed because further IO was attempted after a precondition was violated"
+            let error = ConnectionStateError.preconditionViolated(message: message)
+            self.deliverOneError(context: context, error: error)
```

Comment:
We must not deliver this error: it risks an infinite recursion. It's fine for us to just silently close.